runtime/wasm/src/runtime.rs

```diff
@@ -24,6 +24,19 @@ pub const DATA_TYPE_INPUT: DataTypeIndex = 4;
 pub const DATA_TYPE_RESULT: DataTypeIndex = 5;
 pub const DATA_TYPE_STORAGE_ITER: DataTypeIndex = 6;
 
+/// Because WASM doesn't support u128, we emulate it with 2 u64s.
+type Uint128 = (u64, u64);
```

Comment:
We could avoid arithmetic operations by mapping a pair of u64 directly with u128.
```
type Uint128 = [u64; 2];

#[inline]
fn to_uint128(value: u128) -> Uint128 {
let res = unsafe {
        std::slice::from_raw_parts(&value as *const u128 as *const u64, 2);
    }
[res[0], res[1]]
}

#[inline]
fn from_uint128(value: Uint128) -> u128 {
*unsafe { (&value)[0] as *const u64 as *const u128 as &u128 }
}
```

Reply:
Good point.